Add a multistage recipe

samfoo opened the pull request:
If you refuse this: You obviously don't like burritos. If you don't like burritos, you're broken. QED

samfoo committed:
Simple multistage recipe

quad commented:
"You punched a woman?"
"**I PUNCHED A NAZI!**"

samfoo commented:
Obviously the dynamic stage recipe has to involve superior golden super weapons... made of gold.

samfoo committed:
Rename some methods in stage

"fail" is a keyword and while *technically* "fail?" is a valid method
name it's a bit confusing.

Naming things "fail" and "ok" isn't consistent with our user facing
"FAILED" and "PASSED" verbiage. It might as well be the same, that way
it's less confusing as time goes on.

samfoo committed:
I should have added these previously.

quad commented on doc/cookbook/2-multistage.md:
```diff
@@ -0,0 +1,198 @@
+
+        $ mkdir features
+
+edit a simple feature file `features/burrito_cooking_101.feature`:
```
s/edit/Edit/

quad commented on doc/cookbook/2-multistage.md:
````diff
@@ -0,0 +1,198 @@
+
+And of course, any project needs some files. Using your favorite editor open up `burrito.rb`:
+
+```ruby
````
What happens when GitHub flavoured Markdown hits other Markdown parsers?

samfoo replied:
It looks like shit and breaks syntax highlighting. You prefer code blocks as block quotes? It strikes me a silly to worry about other markdown renderers at the moment.

quad commented on doc/cookbook/2-multistage.md:
```diff
@@ -0,0 +1,198 @@
+
+          1) Burrito#add should add multiple ingredients to a delicious list
+             Failure/Error: burrito.ingredients.should include "queso"
+               expected ["cilantro", "cilantro"] to include "queso"
```
I don't see this as a failure, personally.

quad commented on doc/cookbook/2-multistage.md:
```diff
@@ -0,0 +1,198 @@
+
+## What's Next?
+
+Dynamically stages.
```
I think you accidentally a word.

samfoo committed:
Ready shouldn't mean "not done"

For example:

I am a rockstar developer
I want to be able to run my build multiple times for a given commit
Because sometimes environment failures happen and builds really mean
"green" when they say "red"

Before this commit, any failure broke the build until the next code
commit. This was a bit silly.

By default now, $ bt-go runs all stages again.

samfoo committed:
Start a file with user stories

To be added to as we all become less lazy about docs. (What was that? I
was too busy wailing on my electric guitar to write docs?...)

samfoo committed:
Removed a bunch of cached gems.

Actually this *must* have been why cloning bt takes forever.

samfoo committed:
Add Gemfile.lock and vendor/cache to ignore.

samfoo committed:
Fixing changes for pull #61

samfoo committed:
Vanity
